include/boost/thread/concurrent_queues/sync_timed_queue.hpp

```diff
@@ -229,7 +229,7 @@ namespace detail
       if (super::closed(lk)) return true; // closed
 
       const time_point tp(super::data_.top().time);
-      super::wait_until_closed_until(lk, tp);
```

Comment:
The two changes in this file are what actually fix the issue.

Reply:
Thanks a lot for the quick fix.